curp/src/server/curp_node.rs

```diff
@@ -517,59 +522,56 @@ impl<C: 'static + Command> CurpNode<C> {
                     Ok(req) => req,
                 }
             };
-            // send append_entries to each server in parallel
-            for connect in connects.values() {
-                let _handle = tokio::spawn(Self::send_log_until_succeed(
+            for connect in connects.values().cloned() {
+                let _handle = tokio::spawn(Self::send_log_to_follower(
                     Arc::clone(&curp),
-                    Arc::clone(connect),
+                    connect,
                     i,
                     req.clone(),
                 ));
             }
-            curp.opt_out_hb();
         }
     }
 
     /// Send `append_entries` containing a single log to a server
     #[allow(clippy::integer_arithmetic, clippy::indexing_slicing)] // log.len() >= 1 because we have a fake log[0], indexing of `next_index` or `match_index` won't panic because we created an entry when initializing the server state
-    async fn send_log_until_succeed(
+    async fn send_log_to_follower(
         curp: Arc<RawCurp<C>>,
         connect: Arc<impl ConnectApi>,
         i: LogIndex,
         req: AppendEntriesRequest,
     ) {
-        let (rpc_timeout, retry_timeout) = (curp.cfg().rpc_timeout, curp.cfg().retry_timeout);
+        debug!("{} sends log[{i}] to follower {}", curp.id(), connect.id());
+        let rpc_timeout = curp.cfg().rpc_timeout;
         // send log[i] until succeed
-        loop {
-            let resp = connect.append_entries(req.clone(), rpc_timeout).await;
-
-            #[allow(clippy::unwrap_used)]
-            // indexing of `next_index` or `match_index` won't panic because we created an entry when initializing the server state
-            match resp {
-                Err(e) => {
-                    warn!("append_entries error: {e}");
-                }
-                Ok(resp) => {
-                    let resp = resp.into_inner();
-
-                    let result = curp.handle_append_entries_resp(
-                        connect.id(),
-                        Some(i),
-                        resp.term,
-                        resp.success,
-                        resp.hint_index,
-                    );
+        let resp = connect.append_entries(req, rpc_timeout).await;
```

Comment:
In the original version, the `send_log_until_success` would retry `append_entries` infinitely until it succeeded, whereas there is no retry in the current version. Is this modification too extreme? I think it would be more friendly to specify a maximum number of retries, like 3.

Reply:
Removing retrying here does not mean that we will not retry. Actually, we will retry indefinitely in calibration task